src/pal/inc/rt/palrt.h

```diff
@@ -280,7 +280,7 @@ inline void *__cdecl operator new(size_t, void *_P)
 #define THIS_
 #define THIS                void
 
-#if _MSC_VER
+#if  defined(_MSC_VER) || defined(PAL_STDCPP_COMPAT)
```

Comment:
Why is the defined(PAL_STDCPP_COMPAT) added here? That means that it would use __declspec(xxx) builds using non-MSVC compilers. Do these __declspecs work in clang on Unix?